### PR TITLE
Show edit data and get category list

### PR DESCRIPTION
**Pull request by ghost**
## Description
- get category list and display it into selector of edit from
- get book data with select id and fill data into an edit form
## Reference Requirements
- Product Backlog ID: #PBI-4
- Task ID: #4.2

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- app/Http/Controllers/Admin/BookController.php
- public/js/main.js
- resources/views/backend/books/edit.blade.php
- config/image.php
- config/define.php

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
query1
query2

pic3

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: resources/views/backend/books/edit.blade.php [hunk 1]
```diff
@@ -27,47 +27,56 @@
...
                   <span class="text-danger">{{ $errors->first('name') }}</span>
                 @endif
               </div>
               <div class="form-group">
                 <label for="category">{{ __('book.category') }}</label>
                 <select name="category_id" id="category">
+                  @foreach($categories as $category)
+                    @if($category->id == config('define.default_category'))
+                      continue;
+                    @elseif($book->category_id == $category->id)
```
Comment (at-annguyen) [inline resources/views/backend/books/edit.blade.php]: $book->category->id

File: resources/views/backend/books/edit.blade.php [hunk 2]
```diff
@@ -27,47 +27,56 @@
...
+                      <option value="{{ $category->id }}" selected>{{ $category->name }}</option>
+                    @else
```
Comment (at-annguyen) [inline resources/views/backend/books/edit.blade.php]: delete

File: app/Http/Controllers/Admin/BookController.php [hunk 3]
```diff
@@ -34,4 +33,22 @@ public function index()
...
     }
+
+    /**
+     * Show the form with book data for edit book.
+     *
+     * @param int $id book
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
```
Comment (at-annguyen) [inline app/Http/Controllers/Admin/BookController.php]: u should get object param edit(Book $book)

File: resources/views/backend/books/edit.blade.php [hunk 4]
```diff
@@ -27,47 +27,56 @@
...
                   <span class="text-danger">{{ $errors->first('name') }}</span>
                 @endif
               </div>
               <div class="form-group">
                 <label for="category">{{ __('book.category') }}</label>
                 <select name="category_id" id="category">
+                  @foreach($categories as $category)
+                    @if($category->id == App\Model\Book::DEFAULT_CATEGORY)
+                      continue;
+                    @elseif(old('category_id', $book->category_id) == $category->id)
```
Comment (at-annguyen) [inline resources/views/backend/books/edit.blade.php]: get categories by relationship

File: resources/views/backend/books/edit.blade.php [hunk 5]
```diff
@@ -27,47 +27,56 @@
...
+                      <option value="{{ $category->id }}" selected>{{ $category->name }}</option>
+                    @else
+                      <option value="{{ $category->id }}">{{ $category->name }}</option>
+                    @endif
```
Comment (at-annguyen) [inline resources/views/backend/books/edit.blade.php]: unnecessary

File: app/Http/Controllers/Admin/BookController.php (hunk 3 above)
Comment (at-anhngo) [reply app/Http/Controllers/Admin/BookController.php]: không cần thiết

Comment (at-annguyen) [reply app/Http/Controllers/Admin/BookController.php]: u set param id and then u call Book object where id = $id???? why don't u set param = object???

**Comment by at-annguyen** [comment]
LGTM

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**